core/src/autogluon/core/models/ensemble/fold_fitting_strategy.py

```diff
@@ -581,7 +601,20 @@ def terminate_all_unfinished_tasks(self, unfinished_tasks):
         for task in unfinished_tasks:
             self.ray.cancel(task, force=True)
 
-    def _fit(self, model_base_ref, X_ref, y_ref, X_pseudo_ref, y_pseudo_ref, time_limit_fold, fold_ctx, resources, kwargs):
+    def _fit(
+        self,
+        *,
+        model_base_ref,
+        X_ref,
+        y_ref,
+        X_pseudo_ref,
+        y_pseudo_ref,
```

Comment:
Is there a reason these aren't just kwargs? Currently if I added some new fit arg like `X_unlabeled`, I would need to edit this code (at least I think) for it to work in distributed. Is there a way we can avoid this and make the logic more general?

Reply:
This function is being shared by distributed and local parallel folding. You would need to edit this function anyway to make it work locally.

Reply:
ok, how about if I remove the comment about distributed?

Is there a reason these aren't just kwargs? Is there a way we can avoid this and make the logic more general? For example, `X_val_ref` isn't present here.

Reply:
nvm on X_val_ref, I see you use `fold` to get that

Reply:
deep dived the code, I see the reasoning for how it is structured, disregard this comment.